clair/clair.go

```diff
@@ -89,29 +92,48 @@ func newLayer(image *docker.Image, index int) *layer {
 	}
 }
 
+func filterEmptyLayers(fsLayers []docker.FsLayer) (filteredLayers []docker.FsLayer) {
+	for _, layer := range fsLayers {
+		if layer.BlobSum != EMPTY_LAYER_BLOB_SUM {
+			filteredLayers = append(filteredLayers, layer)
+		}
+	}
+	return
+}
+
 // Analyse sent each layer from Docker image to Clair and returns
 // a list of found vulnerabilities
 func (c *Clair) Analyse(image *docker.Image) []Vulnerability {
+	// Filter the empty layers in image
+	image.FsLayers = filterEmptyLayers(image.FsLayers)
+	layerLength := len(image.FsLayers)
+	if layerLength == 0 {
+		fmt.Printf("No need to analyse image %s/%s:%s as there is no non-emtpy layer",
+			image.Registry, image.Name, image.Tag)
+		return nil
+	}
+
 	var vs []Vulnerability
-	for i := range image.FsLayers {
+	for i := layerLength - 1; i >= 0; i-- {
```

Comment:
I thought you want to analyze just the top level, in code I see that empty layers a filtered out and we reversed the direction of iteration. Did I miss something?